### PR TITLE
Finish off Get imposter feature

### PR DESCRIPTION
**Pull request by nrjohnstone**
Can now assert against calls to imposter.

### CODE REVIEW

**Comment by nrjohnstone** [comment]
This will implement https://github.com/mattherman/MbDotNet/issues/15

File: MbDotNet.Acceptance.Tests/AcceptanceTests/CanVerifyCallsOnImposter.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+        private void CallImposter()
+        {
+            var restClient = new RestClient("http://localhost:6000");
+
+            var request1 = new RestRequest("customers/123", RestSharp.Method.POST);
+            request1.AddBody(new TestData("Bob", "bob@zmail.com"));
+            var response1 = restClient.Execute(request1);
+
+            response1.StatusCode.Should().Be(HttpStatusCode.OK);
```
Comment (mattherman) [inline MbDotNet.Acceptance.Tests/AcceptanceTests/CanVerifyCallsOnImposter.cs:57]: Please rename these to just request and response.

File: MbDotNet/MountebankRequestProxy.cs [hunk 2]
```diff
@@ -90,13 +92,17 @@ private HttpResponseMessage ExecutePost(string resource, string json)
...
         {
+            if (exceptionFactory == null)
+                exceptionFactory = (message) => new MountebankException(message);
+
             if (response.StatusCode != expectedStatusCode)
             {
                 var content = response.Content.ReadAsStringAsync().Result;
                 var errorMessage = $"{failureErrorMessage}\n\nError Message => \n{content}";
-                throw new MountebankException(errorMessage);
+                throw exceptionFactory(errorMessage);
```
Comment (mattherman) [inline MbDotNet/MountebankRequestProxy.cs:105]: Nice  👍

File: MbDotNet/Models/Imposters/Requests.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+using MbDotNet.Enums;
+using Newtonsoft.Json;
+
+namespace MbDotNet.Models.Imposters
+{
+    public class Requests
```
Comment (mattherman) [inline MbDotNet/Models/Imposters/Requests.cs:8]: Please rename this class "Request"

**Comment by mattherman** [comment]
Overall, it looks really good.

One thing I noticed looking at the [documentation](http://www.mbtest.org/docs/api/mocks) for verifying mocks is that the requests object differs depending what type of imposter it was. I think the Request object you created would work for HTTP imposters, but not for TCP or SMTP. I am fine with only supporting one type for now, but we should make this clearer. 

We should also set it up so that it can be extended for TCP/SMTP, preferably without breaking changes.
I'm not sure right now how to handle the multiple Request types, but I'll think about it. Open to suggestions as well. ( #18 )

I'll merge this quick and fix my two little comments so you don't have to bother with it. Thanks.

**Pull request merged**